### DISCUSSION TITLE
Converted network from pandapower does not converge but the same network in pandapower converges

### DISCUSSION THREAD

**Discussion by AnkurArohi** [PowerGridModel/.github#40] [Q&A] [answered]
Hi PGM team, 
I am trying to evaluate what PGM can offer, in this process we have a pandapower network which we are converting to PGM following the instructions in the documentation. The network is properly converted but then when we execute the load flow calculation we have this error
Here are the details of the network model keys
{'_model_ptr': <ModelPtr object at 0x7fc4cce8f7d0>, '_all_component_count': {<ComponentType.node: 'node'>: 12879, <ComponentType.line: 'line'>: 1073, <ComponentType.source: 'source'>: 1, <ComponentType.sym_load: 'sym_load'>: 1485, <ComponentType.shunt: 'shunt'>: 151, <ComponentType.transformer: 'transformer'>: 650, <ComponentType.sym_gen: 'sym_gen'>: 685, <ComponentType.three_winding_transformer: 'three_winding_transformer'>: 226, <ComponentType.link: 'link'>: 11956}}
[debug    ] 2. Native Power Grid Model error: Field 'tap_size' is not greater than (or equal to) zero for 203 ComponentsTypes.three_winding_transformers.
 [critical ] Internal SparseMatrixError occurred!
 [debug    ] Sparse matrix error, possibly singular matrix!
If you get this error from state estimation, it usually means the system is not fully observable, i.e. not enough measurements.
Try validate_input_data() or validate_batch_data() to validate your data.

**Comment by TonyXiang8787** [comment]
Hi @AnkurArohi,

Thanks for your question. Can you upload the pandapower network and converted PGM network, so we can have a look?

For pandapower you can use their export function:
https://pandapower.readthedocs.io/en/latest/file_io.html#json

For PGM you can use our export function:
https://power-grid-model.readthedocs.io/en/stable/examples/Serialization%20Example.html

**Reply by AnkurArohi** [reply]
sorry the network cannot be exported due to confidentiality

**Comment by mgovers** [comment, marked as answer]
Hi @AnkurArohi ,

Thank you very much for checking out our project!

There are many reasons why a sparse matrix error can occur, ranging from bad/missing input data to coincidental underdetermined edge cases. Seeing that you have this line in your log, it appears that it's the former, because any invalid data may result in unspecified behavior in the PGM calculations (like sparse matrix errors):

> ```
> [debug    ] 2. Native Power Grid Model error: Field 'tap_size' is not greater than (or equal to) zero for 203 ComponentsTypes.three_winding_transformers.
> ```

Note that this error comes from the data validator and it complains specifically about the fact that your three-winding transformer input is not valid: tap_size needs to be >=0 but is not in your converted input data (see also https://power-grid-model.readthedocs.io/en/stable/user_manual/components.html#id14).

If I look at the most recent implementation in power-grid-model-io, which I expect you use to convert your pandapower data to PGM data (https://github.com/PowerGridModel/power-grid-model-io/blob/main/src/power_grid_model_io/converters/pandapower_converter.py#L804-L938 and https://github.com/PowerGridModel/power-grid-model-io/blob/main/src/power_grid_model_io/converters/pandapower_converter.py#L2304-L2326), then it would appear that tap_size is only set to a value <0 if and only if any of your tap_step_percent, vn_hv_kv, vn_mv_kv or vn_lv_kv are <0 for some of your three-winding transformers. Other values are either >=0 by itself, or defaulted to 0. Can you please verify whether there are such three-winding transformers in your input data?

NOTE: I don't know whether the line about the three-winding transformer is the only log entry concerning data validation errors. There might be other ones as well, so please double-check. As I mentioned before, any data validation error may result in unspecified behavior like triggering sparse matrix errors.

Hope this helps. If not, I'd be happy to dive into this further together with you.

Kind regards,

Martijn

**Reply by AnkurArohi** [reply]
I will check the tap_pos , tap_max, tap_min for 3W trafos
